protocol/x/clob/keeper/msg_server_cancel_orders.go

```diff
@@ -22,6 +26,20 @@ func (k msgServer) CancelOrder(
 
 	defer func() {
 		if err != nil {
+			// Gracefully handle the case where the order was already removed from state.
+			if errors.Is(err, types.ErrStatefulOrderDoesNotExist) {
+				processProposerMatchesEvents := k.Keeper.GetProcessProposerMatchesEvents(ctx)
+				removedOrderIds := lib.SliceToSet(processProposerMatchesEvents.RemovedStatefulOrderIds)
+				if _, found := removedOrderIds[msg.GetOrderId()]; found {
+					k.Keeper.Logger(ctx).Info(
+						errorsmod.Wrap(
+							err,
+							"MsgCancelOrder failed DeliverTx because the order was already removed from state",
```

Comment:
Can you include the order ID in the log?

Reply:
the order id is included in the original error returned from PerformOrderCancellationStatefulValidation